File: code_examples/3-Structured_Problem_Solving/16-Break/Go/main.go
/**
 * Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a break statement
 */

package main

import (
	"fmt"
)

func main() {
	// this function uses a break statement

	var counter int // in go no need to set to 0, it automaticall is!
	var positiveInteger int

	// input
	fmt.Print("Enter how many times to repeat: ")
	fmt.Scan(&positiveInteger)
	fmt.Println()

	// process & output
	for counter < positiveInteger {
		fmt.Printf("%d time(s) through the loop.\n", counter)
		if counter >= 5 {
			break
		}
		counter++ // a short form for: counter = counter + 1
	}

	fmt.Println("\nDone.")
}

File: code_examples/3-Structured_Problem_Solving/16-Break/JavaScript/main.js
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program uses a break statement
 */

const prompt = require("prompt-sync")()

let counter = 0

// input
const positiveIntegerStr = prompt("Enter how many times to repeat: ")
console.log("")

// process & output
const positiveInteger = parseInt(positiveIntegerStr)

for (var loopCounter = 0; loopCounter < positiveInteger; loopCounter++) {
  console.log(`${counter} time(s) through the loop.`)
  if (loopCounter >= 5){
    break
  }
}

console.log("\nDone.")
